packages/cached/src/cached-fs.ts

```diff
@@ -32,9 +32,17 @@ export function createCachedFs(fs: IFileSystem): ICachedFileSystem {
   const realpathCache = new Map<string, string>();
   const { promises } = fs;
   const invalidateAbsolute = (absolutePath: string) => {
-    const cachePath = getCanonicalPath(absolutePath);
-    realpathCache.delete(cachePath);
-    statsCache.delete(cachePath);
+    const cachePathPrefix = getCanonicalPath(absolutePath);
+    for (const key of realpathCache.keys()) {
```

Comment:
in cases where we change x, x.y would also be invalidated, we need to find a way to make sure we don't invalidate things that shouldn't be invalidated